gcc/config/riscv/riscv.c

```diff
@@ -4583,6 +4586,21 @@ riscv_option_override (void)
     error ("%<-mriscv-attribute%> RISC-V ELF attribute requires GNU as 2.32"
 	   " [%<-mriscv-attribute%>]");
 #endif
+
+  /* If user don't set the `m[no-]check-constraints` option and assembler
+     support the constraints checking, then GCC should enable it by default.
+     If user set the `mcheck-constraints` options but the checking isn't
+     supported in assembler, then we should report error here.  */
+  if (riscv_mcheck_constraints < 0)
+#ifdef HAVE_AS_RISCV_CHECK_CONSTRAINTS
+    riscv_mcheck_constraints = 1;
+#else
+    riscv_mcheck_constraints = 0;
+
+  if (riscv_mcheck_constraints)
+    error ("%<-mcheck-constraints%> RISC-V constraints checking requires GNU"
+	   " as 2.33.1 [%<-mcheck-constraints%>]");
```

Comment:
Giving the gas version number in the error message looks a little odd, as this feature isn't in upstream GNU as, upstream GNU as is already to 2.34, and is only in vector branch with the riscv-binutils-gdb  #193 pr which doesn't change the gnu as version number.  I think it is better without version number until the patches are upstream and there is a proper binutils release we can refer to.

Reply:
OK, I update the message as follows,
"RISC-V constraints checking is unsupported in GNU as"